org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/model/ContentModelManager.java

```diff
@@ -310,6 +310,10 @@ private CMDocument findCMDocument(String uri, String publicId, String systemId,
 				Path file = cacheResolverExtension.getCachedResource(resolvedUri);
 				if (file != null) {
 					cmDocument = modelProvider.createCMDocument(file.toUri().toString());
+				} else {
+					// It's not a file URL, and not a remote URL (ftp, http, https)? It's something else that might have a registered URL Handler.
+					// Therefore just try to load the model from the URL.
+					cmDocument = modelProvider.createCMDocument(resolvedUri);

```

Comment:
I understand now why you did that (thanks for your test!) but I think it's not a clean fix. I think a correct fix should provide the capability to define protocol which benefit with cache or not.

Let me find an idea that I will try to suggest it in order to we can discuss together (if you like it or not).

Reply:
@BoykoAlex I create a PR  https://github.com/eclipse/lemminx/pull/907 please review it and tell me if it works for you. Thanks!

Reply:
Yes, it would work for me as well, lets go with #907 and I'll close this one